### ISSUE TITLE
No completion for `export import`

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#26540]
**TypeScript Version:**  3.1.0-dev.20180818

**Code**

```ts
declare global {
    namespace N {
	    const foo: number;
    }
}

export import foo = N.foo;

/**/
```

Start typing `foo`.

**Expected behavior:**

Get a completion for `foo`.

**Actual behavior:**

No such completion.

**Comment by noahbkim** [comment]
Where should I be looking for completion code? And how can I test completion?

**Comment by DanielRosenwasser** [comment]
Hey Noah! You'll want to look for `getCompletionsAtPosition` in `services` (I believe the file is called `completions.ts`).

From there, you'll want to write a test case in `test/cases/fourslash` - there should be other tests around completions in that folder for you to try. [Read up on the `fourslash.ts` file in that folder.](https://github.com/Microsoft/TypeScript/blob/f2a1a428ea99ace495a8ca769b0a7c857ea6486a/tests/cases/fourslash/fourslash.ts#L1-L27). Basically, a single marker is indicated with a `/**/` comment, and you can use the imperative APIs like `goTo.marker()` and `verify.completionsAtPosition(...)` to write the test.

**Comment by jrkong** [comment]
@noahbkim are you still looking at this issue? If not I would like to take a stab at it.

**Comment by noahbkim** [comment]
@jrkong go for it.

**Comment by jrkong** [comment]
@noahbkim Thanks!

I've been doing a bit of digging beforehand and so far I've found that the `export` keyword takes priority and trumps `import` so no completion is displayed. Can I assume that exported variables should not appear on the completion list?

Depending on the answer my hypothetical fix would be to:
- add exported variables and functions to the completion list
or
- add an additional case when export is detected to check if it is followed by an import, and if that's the case then add to the completion list

Working under those assumptions I've been trying to find the part of the code that generates the completions when an import is detected. I believe it is https://github.com/Microsoft/TypeScript/blob/12cd3ea6ac1fd9475001741ef1b7278d69540bd8/src/services/completions.ts#L407-L408 but I'm not sure.

Am I headed in the right direction?
